extensions/azure/storage/AzureBlobStorage.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file AzureBlobStorage.cpp
+ * AzureBlobStorage class implementation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "AzureBlobStorage.h"
+#include "utils/GeneralUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace azure {
+namespace storage {
+
+AzureBlobStorage::AzureBlobStorage(const std::string &connection_string, const std::string &container_name)
+  : BlobStorage(connection_string, container_name)
+  , container_client_(minifi::utils::make_unique<Azure::Storage::Blobs::BlobContainerClient>(
+      Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(connection_string, container_name))) {
+}
+
+void AzureBlobStorage::resetClientIfNeeded(const std::string &connection_string, const std::string &container_name) {
+  if (connection_string == connection_string_ && container_name_ == container_name) {
+    logger_->log_debug("Client credentials have not changed, no need to reset client");
+    return;
+  }
+  connection_string_ = connection_string;
+  container_name_ = container_name;
+  logger_->log_debug("Client has been reset with new credentials");
+  container_client_ = minifi::utils::make_unique<Azure::Storage::Blobs::BlobContainerClient>(Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(connection_string, container_name));
```

Comment:
Minor, but calling `::reset` would be a little bit less boilerplate.

Reply:
`CreateFromConnectionString` does not return a pointer but an object so we cannot use `reset` here.